gdal/alg/viewshed.cpp

```diff
@@ -255,17 +250,14 @@ CPLErr GDALViewshedGenerate(GDALRasterBandH hBand, const char* pszTargetRasterNa
         return CE_Failure;
     }
 
-    double *padfFirstLineVal = static_cast<double *>(
-        vFirstLineVal.data());
-    double *padfLastLineVal = static_cast<double *>(
-        vLastLineVal.data());
-    double *padfThisLineVal = static_cast<double *>(
-        vThisLineVal.data());
-    GByte *pabyResult = static_cast<GByte *>(
-        vResult.data());
+    double *padfFirstLineVal = vFirstLineVal.data();
+    double *padfLastLineVal = vLastLineVal.data();
+    double *padfThisLineVal = vThisLineVal.data();
+    GByte *pabyResult = vResult.data();
 
     GDALRasterBandH hTargetBand = nullptr;
-    GDALDatasetH hDstDS = nullptr;
+    std::unique_ptr<GDALDataset> hDstDS{};
```

Comment:
poDstDS would probably be best regarding naming convention. We don't really have a naming convention for smart pointers. or just dstDS perhaps